lib/mpool/src/mcache.c

```diff
@@ -30,7 +30,7 @@ struct mpool_mcache_map {
     struct mpool *mp;
     size_t        mbidc;
     uint64_t     *mbidv;
-    void        **addrv;
+    const void  **addrv;
```

Comment:
Just leave this as the same type we get back from mmap to avoid all those unnatural contortions back to non-const.

Reply:
Unnatural contortions (i.e., casts) are unavoidable in this situation. We can declare `addrv` as a `const` in our code and cast when making system calls (e,g, `mprotect` and `madvise`), or declare `addrv` as a `non-const` and cast when we want to hand `addrv[i]` back to a caller (e.g., in `mpool_mcache_getbase()`).  I think it's cleaner to declare it `const` and cast as required by APIs that we don't control.

Reply:
I totally disagree, the presence of casting away constness is a huge red flag which makes me have to go review a bunch of code to figure out if it's safe, an unnecessary exercise otherwise...
What's the harm in leaving the underlying implementation as is?  By changing it you now have to go de-constify a bunch of use cases, that alone should have set off alarms...

Reply:
I don't think this one needs to be const, since it is in the .c file. When the caller asks for an addr, then we could return a const pointer.